src/Microsoft.ServiceFabric.Client/ClientSettings.cs

```diff
@@ -44,5 +44,10 @@ public class ClientSettings
         /// Gets the Timespan to wait before the request times out for the client.
         /// </summary>
         public TimeSpan? ClientTimeout { get; }
+
+        /// <summary>
+        /// Sets the connection string to use for the image store. If not set, the connection string will be read from the cluster
+        /// </summary>
+        public string ImageStoreConnectionString { get; set; }
```

Comment:
Client Settings contains general settings for the client. ImageStoreConnectionString is specific only to few APIs. Lets not expose it from here and do it via corresponding method overloads.

Reply:
So it should be a property on IImageStore?
That mean the Imagestore implementation would have to call the client and sort of replace it self. that's fine i suppose. i would prefer to keep the file and service implementations separate rather than having if statements for each kind in a single implementation

Reply:
In this case if statement in Upload and delete methods would be better. ImageStoreClient doesn't need to determine the connection string unless a method to Upload and Delete is called on it, then it can just cache that value.

Reply:
But you've said that the class implementations are auto generated, then how can I modify the upload and delete methods without them being overwritten?

Reply:
The Upload method was in the IApplicationClient_ which was not an autogenerated method as there is no such REST API which can upload the application package as such. So you can definitely add it. For Delete you can make the change to it, i will look into special casing it during code generation or in post processing.

Reply:
allright, i'll add if statements to the other applicable methods (like listing, those are also useful)